### PR TITLE
[Home Page Picker] No Network handling

### PR DESCRIPTION
**Pull request by chipsnyder**
**Fixes** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2718 (No Network)

📓 This is based on the changes in https://github.com/wordpress-mobile/WordPress-iOS/pull/15213
 
## To test:
<details>
<summary><strong>To disable or enable the development version of Home Page Picker</strong></summary>

- Open the app from the build that allows FeatureFlags such as a PR build or a local development build
    - By default, the modal layout picker will be disabled.
    - From the site page:
        - Click on your Gravatar.
        - Click on App Settings.
        - Click on Debug.
        - Toggle "Site Creation: Home Page Picker" to enable or disable the picker
        - <kdb><a href="https://user-images.githubusercontent.com/3384451/97343011-5f032980-185d-11eb-9030-06c17f732fec.gif"></a></kdb>

</details>

When enabled, the first step of the Site Creation flow will be replaced with the Home Page Picker flow.

### No Network
1. Disable your network connection
1. Navigate to the Create WordPress.com site
    - My Sites > ➕ > Create WordPress.com site
    - **Expect** to see the No Network Screen
1. Enable your network connection
1. Select Retry
    - **Expect** to see the data load and the no network prompt to go away.

## Screenshots:
<kdb><a href="https://user-images.githubusercontent.com/3384451/98157948-29d38880-1ea8-11eb-9d28-114b1f65a9cc.png"></a></kdb>

## Additional Context: 

This is a continuation of the work done in the [Home Page Picker Project](https://github.com/wordpress-mobile/gutenberg-mobile/labels/Project%3A%20Home%20Page%20Picker). 

## PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Site Creation/DesignSelection/SiteDesignContentCollectionViewController.swift [hunk 1]
```diff
@@ -114,7 +117,9 @@ class SiteDesignContentCollectionViewController: CollapsableHeaderViewController
     }
 
     private func handleError(_ error: Error) {
-        /* ToDo */
+        let titleText = NSLocalizedString("Unable to load this content right now.", comment: "Informing the user that a network request failed becuase the device wasn't able to establish a network connection.")
+        let subtitleText = NSLocalizedString("Check your network connection and try again or create a blank page.", comment: "Default subtitle for no-results when there is no connection with a prompt to create a new page instead.")
```
Comment (mkevins) [inline WordPress/Classes/ViewRelated/Site Creation/DesignSelection/SiteDesignContentCollectionViewController.swift]: Do we want this to say "or create a blank page"?

**Review by mkevins — COMMENTED** [review]
Tested and this works as expected. Just one question about the wording of the error.

Nice work on this Chip! :+1:

File: WordPress/Classes/ViewRelated/Site Creation/DesignSelection/SiteDesignContentCollectionViewController.swift (hunk 1 above)
Comment (chipsnyder) [reply WordPress/Classes/ViewRelated/Site Creation/DesignSelection/SiteDesignContentCollectionViewController.swift]: Oops probably not haha

**Review by mkevins — APPROVED** [review]
Nice work! LGTM :tada:

**Review by antonis — APPROVED** [review]
Nice work Chip 👍 
LGTM 🎉

**Pull request merged**